### PR TITLE
feat: create GET API request for /api/users endpoint

### PR DESCRIPTION
**Pull request by MjxOro**
# Summary
This PR I created the GET API request for the `/api/users` endpoint. This endpoint returns the current user object data. The controller I created, grabs the jwt token from the Authorization header. That token is verified and decoded to use the decoded value to query the user by their email and return that data back to the requester. The format of the code returned:
``` JSON
{
  "user": {
    "email": "jake@jake.jake",
    "token": "jwt.token.here",
    "username": "jake",
    "bio": "I work at statefarm",
    "image": null
  }
}

```
# Screen shot of the tests
Screenshot 2022-10-07 at 7 48 33 PM

### CODE REVIEW

File: src/controllers/users.js [hunk 1]
```diff
@@ -29,33 +30,44 @@ export async function registerUser(req, res) {
       email: user.email,
       hash: hash,
     });
-    const token = signToken(newUser)
-    newUser.setDataValue('token', token)
-    await newUser.save()
-    const responseData = userResponse(newUser)
+    const token = signToken(newUser.get());
+    newUser.setDataValue("token", token);
```
Comment (YoussefLaunchUp) [inline src/controllers/users.js]: Lets rethink how we are using the JWT tokens. We should not store them in the db, instead we can set an expiry time and during this time we know that the user is logged in. 
Have a look at this for more info: https://stackoverflow.com/questions/42763146/does-it-make-sense-to-store-jwt-in-a-database

File: src/controllers/users.js [hunk 2]
```diff
@@ -29,33 +30,44 @@ export async function registerUser(req, res) {
...
     return res.status(201).json(responseData);
   } catch (e) {
     console.error(e);
-    if (e === "No payload found") return res.status(400).send("No payload found")
-    if (e === "Invalid payload format") return res.status(400).send("Invalid payload format, refer to docs for payload format")
-    if (e === "Invalid email format") return res.status(400).send("Invalid email format")
-    return res.status(500).send("Server error")
+    if (e.message === "No payload found")
+      return res.status(400).send(e.message);
+    if (e.message === "Invalid payload format")
```
Comment (YoussefLaunchUp) [inline src/controllers/users.js]: Lets combine these if statements for readability.

**Review by YoussefLaunchUp — CHANGES_REQUESTED** [review]
Some changes with JWT are required, other than that good job!

File: src/controllers/users.js (hunk 1 above)
Comment (MjxOro) [reply src/controllers/users.js]: Interesting.  I will change jwt strategy.

Comment (MjxOro) [reply src/controllers/users.js]: I added an expiration to the token and wont be stored in the database

File: src/controllers/users.js (hunk 2 above)
Comment (MjxOro) [reply src/controllers/users.js]: I collected the errors into 1 if statement

File: src/controllers/users.js [hunk 3]
```diff
@@ -21,29 +20,22 @@ export async function registerUser(req, res) {
...
     return res.status(201).json(responseData);
   } catch (e) {
     console.error(e);
-    if (e.message === "No payload found")
-      return res.status(400).send(e.message);
-    if (e.message === "Invalid payload format")
-      return res.status(400).send(e.message);
-    if (e.message === "Invalid email format")
+    if (
+      e.message === "No payload found" ||
```
Comment (YoussefLaunchUp) [inline src/controllers/users.js:31]: nit: we can move the error messages into an array or object and just check if e.message is in it. Should make it cleaner and easier to change in the future.

File: src/controllers/users.js [hunk 4]
```diff
@@ -52,22 +44,17 @@ export async function getUser(req, res) {
...
-    const decode = jwt.verify(token, process.env.PUBLIC_KEY, { algorithms: 'RS256' })
-
-    const user = await UserModel.findByPk(decode.email)
-
-    const responseData = userResponse(user.get())
-
-    return res.status(200).json(responseData)
-
+    const token = getToken(req.get("Authorization"));
+    const decode = verifyToken(token);
```
Comment (YoussefLaunchUp) [inline src/controllers/users.js:48]: something to think about for the future, there should be an authorization middleware that handles token checks.

**Review by YoussefLaunchUp — APPROVED** [review]
Great work!

File: src/controllers/users.js (hunk 3 above)
Comment (MjxOro) [reply src/controllers/users.js:31]: I'll make a utility file for error messages

File: src/controllers/users.js (hunk 4 above)
Comment (MjxOro) [reply src/controllers/users.js:48]: I'll create a middleware for authorization routes

File: src/controllers/users.js (hunk 3 above)
Comment (MjxOro) [reply src/controllers/users.js:31]: I'll make the updates on the next pull request

File: src/controllers/users.js (hunk 4 above)
Comment (MjxOro) [reply src/controllers/users.js:48]: I'll make the updates on the next pull request

**Pull request merged**